Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -21,6 +21,145 @@
+                                      lz(i, j, k) * dx, lz(i + 1, j, k) * dx});
+#else
+            amrex::Abort("ComputeSStab: Only implemented in 2D3V and 3D3V");
+#endif
+        }
+        else if constexpr(dim == 2){
+            return 0.5_rt * std::max({lx(i, j, k) * dy, lx(i, j + 1, k) * dy,
+                                      ly(i, j, k) * dx, ly(i + 1, j, k) * dx});
+        }
+
+        amrex::Abort("ComputeSStab: dim must be 0, 1 or 2");
+
+        return -1;
+    }
+
+
+    /**
+    * \brief Compute the minimal area for stability for the face i, j, k with normal 'dim'.
+    * (wrapper to allow using ComputeSStab as a non-templated function, by passing 'dim' as an argument)
+    *
+    * \param[in] i, j, k the indices of the cell
+    * \param[in] lx, ly, lz arrays containing the edge lengths
+    * \param[in] dx, dy, dz the mesh with in each direction
+    * \param[in] dim normal direction to the plane in consideration (0 for x, 1 for y, 2 for z)
+    */
+   [[nodiscard]] [[maybe_unused]]
+    AMREX_GPU_DEVICE AMREX_FORCE_INLINE
+    amrex::Real
+    ComputeSStab(const int i, const int j, const int k,
+             const amrex::Array4<const amrex::Real> lx,
+             const amrex::Array4<const amrex::Real> ly,
+             const amrex::Array4<const amrex::Real> lz,
+             const amrex::Real dx, const amrex::Real dy, const amrex::Real dz,
+             const int dim)
+    {
+        if (dim == 0) {
+            return ::ComputeSStab<0>(i,j,k,lx,ly,lz,dx,dy,dz);
+        }
+        else if (dim == 1) {
+            return ::ComputeSStab<1>(i,j,k,lx,ly,lz,dx,dy,dz);
+        }
+        else if (dim == 2) {
+            return ::ComputeSStab<2>(i,j,k,lx,ly,lz,dx,dy,dz);
+        }
+        return -1;
+    }
+
+
+    /**
+    * \brief Whenever an unstable cell cannot be extended we increase its area to be the minimal for stability.
+    *        This is the method Benkler-Chavannes-Kuster method and it is less accurate than the regular ECT but it
+    *        still works better than staircasing. (see https://ieeexplore.ieee.org/document/1638381)
+    *
+    * @tparam      idim Integer indicating the dimension (x->0, y->1, z->2) for which the BCK correction is done
+    *
+    * @param[in]      cell_size_max_lev The cell size at the maximum refinement level
+    * @param[in, out] all_fields The field manager
+    * @param[in]      flag_ext_face The extension flag used by the ECT solver
+    * @param[out]     flag_info_face The info flag used by the ECT solver
+    */
+    template <int idim>
+    void ApplyBCKCorrection(
+        const std::array<amrex::Real,3> &cell_size_max_lev,
+        ablastr::fields::MultiFabRegister& all_fields,
+        const int max_level,
+        const amrex::Vector<std::array< std::unique_ptr<amrex::iMultiFab>, 3 > >& flag_ext_face,
+        const amrex::Vector<std::array< std::unique_ptr<amrex::iMultiFab>, 3 > >& flag_info_face)
```

Comment:
Same suggestion as above (but indentation is fine here):
```suggestion
    void ApplyBCKCorrection (
        const std::array<amrex::Real,3> &cell_size_max_lev,
        ablastr::fields::MultiFabRegister& all_fields,
        const int max_level,
        const amrex::Vector<std::array< std::unique_ptr<amrex::iMultiFab>, 3 > >& flag_ext_face,
        const amrex::Vector<std::array< std::unique_ptr<amrex::iMultiFab>, 3 > >& flag_info_face)
```